airflow/providers/docker/operators/docker.py

```diff
@@ -420,6 +434,8 @@ def _run_image_with_mounts(self, target_mounts, add_tmp_variable: bool) -> list[
         )
         logstream = self.cli.attach(container=self.container["Id"], stdout=True, stderr=True, stream=True)
         try:
+            self._change_log_formatters()
+
```

Comment:
Can you add try/finally clause? what happens if there is an exception thrown inside?

Reply:
Added. Thank you for the review!

Reply:
Hmm. Maybe I am nissing something I thought about try/finally where `_restore_log_formatter` will be called in the finally clause.

Reply:
That makes sense. Updates :)